### DISCUSSION TITLE
ioboker Befehle per mqtt senden

### DISCUSSION THREAD

**Discussion by Seppel9678** [dewenni/ESP_Buderus_KM271#38] [General]
Hallo, 
könnte mir jemand behilflich sein,wie ich Befehle per mqtt von iobroker aus an das Modul sende?

Danke im vorraus

**Comment by dewenni** [comment]
Hi @Seppel9678,

in ioBroker werden erstmal nur die Objekte angelegt die vom Client auch gepublished werden.
Also nach dem start werden von der Logamatic erst alle Config Werte gelesen und dann der Reihe nach die Status Werte.
Diese werden dann per mqtt geschickt und ioBroker legt diese als Objekte an.

Bei dem Befehlen ist jetzt nicht so. Befehle kommen ja üblicherweise von anderen Clients.
Du könntest mal testen in dem du die Befehle über einen anderen mqtt Client absetzt. Z.B. den MQTT-Explorer.
Dann werden die Objekte vermutlich auch in ioBroker angelegt. Nach einem Neustart vom ioBroker sind die dann aber vermutlich auch wieder weg.

Du musst also die Objekte am besten beim Neustart vom ioBroker selbstständig anlegen lassen.
Da scheint es aber auch zuletzt noch ein Thema gegeben zu haben:
https://forum.iobroker.net/topic/46814/admin5-ich-kann-keine-objekte-mehr-manuell-anlegen
https://github.com/ioBroker/ioBroker.admin/issues/1067

Grüße Sven

**Comment by Seppel9678** [comment]
Hab noch mal experimentiert. Das Modul geht in den AP-Mode, sobald mann den Ordner setvalue und die Datenpunkte anlegt.

**Comment by dewenni** [comment]
Das scheint wohl das gleiche Problem zu sein wie hier #36 
Mir fehlt aktuell leider noch das Verständnis was hier passiert.
Passiert das bereits beim Anlegen des ersten Datenpunktes?
Kannst du mit einem MQTT Tool dann mal schauen was da so passiert?
Also was da geschickt wird sobald der Datenpunkt in ioBroker angelegt wird?

**Comment by dewenni** [comment]
OK, ich konnte das nun nachstellen. 100% verstanden habe ich es noch nicht, aber es hat wohl damit zu tun, dass ioBroker beim Verbinden automatisch alle States sendet. Das bringt den ESP dann zum Absturz.

Mit diesen Settings bzw. ohne diese beiden Optionen scheint es aber zu funktionieren:

<img width="1487" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/729057d8-5213-44f9-9f6c-607e65a41bd4">  

Kannst du das mal testen?
Entscheidend ist wohl das deaktivieren von "States bei subscribe publizieren", ich würde aber auch empfehlen die Option "Eigene States beim Verbinden publizieren" zu deaktivieren.

**Reply by Seppel9678** [reply]
Ok. Werde ich heute Abend probieren. Danke.

**Reply by Seppel9678** [reply]
Hallo. 
Funktioniert bis jetzt einwandfrei. 
Hab den Haken bei "States bei subscribe publizieren" rausgenommen.
Tausend Dank für die Arbeit.
